packages/nimble-components/src/select/index.ts

```diff
@@ -471,20 +473,23 @@ export class Select
         this.clearSelection();
         this.filterOptions();
 
-        if (this.filteredOptions.length > 0) {
-            const enabledOptions = this.filteredOptions.filter(
-                o => !o.disabled
-            );
-            if (enabledOptions.length > 0) {
-                enabledOptions[0]!.selected = true;
-            } else {
-                // only filtered option is disabled
-                this.selectedOptions = [];
-                this.selectedIndex = -1;
-            }
-        } else if (this.committedSelectedOption) {
-            this.committedSelectedOption.selected = true;
+        const enabledOptions = this.filteredOptions.filter(o => !o.disabled);
+        let selectedOptionIndex = this.filter !== ''
+            ? this.openActiveIndex ?? this.selectedIndex
+            : this.selectedIndex;
+
+        if (
+            enabledOptions.length > 0
+            && (this.selectedIndex < 0
```

Comment:
What is the purpose of the `this.selectedIndex < 0` clause here? Shouldn't the active option always be updated to something within the filtered list, regardless of whether or not the currently selected item is in the filtered list?

Reply:
Removed this check.